### PR TITLE
scrollTo can be used to scroll to specified selector

### PR DESCRIPTION
**Pull request by Kreozot**
scrollToSelector function + test + readme

### CODE REVIEW

**Comment by matthewmueller** [comment]
Thanks for the PR! Mentioned this in another PR, but I think the API should be:

`nightmare.scrollTo(top, left)`
`nightmare.scrollTo(selector)`

Where scrollTo supports both. What do you think?

**Comment by Kreozot** [comment]
Good point, thank you! Maybe I also should do it in #358, how do you think?

**Comment by matthewmueller** [comment]
@Kreozot haha, yah i was thinking about that... not sure how it'd look unless we break the API. I guess we could maybe do:

```
.screenshot(path, [selector], [, clip])
```

Almost wondering if it just makes more sense to just use an object:

```js
.screenshot({
  path: ...,
  selector: ...,
  clip: ...
})
```

Also, how does this handle multiple elements on different parts of the page by the way?

**Comment by Kreozot** [comment]
It won't breaks the API if we will check argument type.

**Comment by Kreozot** [comment]
@matthewmueller Please take a look.

File: lib/actions.js [hunk 1]
```diff
@@ -386,13 +386,41 @@ exports.useragent = function(useragent, done) {
...
+      var element = document.querySelector(selector);
+      if (element) {
+        var rect = element.getBoundingClientRect();
+        window.scrollTo(Math.round(rect.left), Math.round(rect.top));
+      } else {
+        return 'invalid selector "' + selector + '"';
+      }
+    }, function (a, error) {
+      if (error) {
+        throw new Error(error);
```
Comment (matthewmueller) [inline lib/actions.js]: this should probably be: `done(error)`

File: lib/actions.js [hunk 2]
```diff
@@ -386,13 +386,41 @@ exports.useragent = function(useragent, done) {
...
+  }
+
+  if (selector) {
+    this._evaluate(function (selector) {
+      var element = document.querySelector(selector);
+      if (element) {
+        var rect = element.getBoundingClientRect();
+        window.scrollTo(Math.round(rect.left), Math.round(rect.top));
+      } else {
+        return 'invalid selector "' + selector + '"';
```
Comment (matthewmueller) [inline lib/actions.js]: i believe the throwing should be done here

File: lib/actions.js [hunk 3]
```diff
@@ -386,13 +386,41 @@ exports.useragent = function(useragent, done) {
...
+      }
+    }, function (a, error) {
```
Comment (matthewmueller) [inline lib/actions.js]: what's "a"?

**Comment by highvoltag3** [comment]
@rosshinkley just wondering is there any reason for this PR's to still be awaiting merge?

**Comment by rosshinkley** [comment]
@highvoltag3  I'm still open to this PR, but the source is well out of date and would need to be updated.  Additionally, I don't think @matthewmueller's comments were ever addressed.

**Pull request closed without merging**